code/modules/mob/living/simple_animal/hostile/necro.dm

```diff
@@ -96,11 +130,318 @@
+			stop_automated_movement = 0
+			walk(src,0)
+/mob/living/simple_animal/hostile/necro/zombie/proc/can_open_door(var/obj/machinery/door/D)
+	if(istype(D,/obj/machinery/door/poddoor) || istype(D, /obj/machinery/door/airlock/multi_tile/glass) || istype(D, /obj/machinery/door/window))
+		return 0
+
+	// Don't fuck with doors that are doing something
+	if(D.operating>0)
+		return 0
+
+	// Don't open opened doors.
+	if(!D.density)
+		return 0
+
+	// Can't open bolted/welded doors
+	if(istype(D,/obj/machinery/door/airlock))
+		var/obj/machinery/door/airlock/A=D
+		if(A.locked || A.welded || A.jammed)
+			if(break_doors == CANPLUS)
+				return 1
+			else
+				return 0
+
+	return 1
+
+/mob/living/simple_animal/hostile/necro/zombie/proc/force_door(var/obj/machinery/door/D)
+	var/time_mult = 1
+	if(istype(D, /obj/machinery/door/airlock/))
+		var/obj/machinery/door/airlock/A = D
+		if(A.locked)
+			time_mult += 1
+		if(A.welded)
+			time_mult += 1
+		if(A.jammed)
+			time_mult += 1
+	stop_automated_movement = 1
+	D.visible_message("<span class='warning'>\The [D]'s motors whine as something attempts to brute force their way through it!</span>")
+	playsound(get_turf(D), 'sound/effects/grillehit.ogg', 50, 1)
+	D.shake(1, 8)
+	if(do_after(src, D, 100*time_mult, needhand = FALSE)) //Roughly 10 seconds for people on the other side of the door to run the heck away
+		if(can_open_door(D))//Let's see if nobody quickly bolted it
+			if(break_doors == CANPLUS) //Guaranteed
+				D.visible_message("<span class='warning'>\The [D] breaks open under the pressure</span>")
+				if(istype(D, /obj/machinery/door/airlock/))
+					var/obj/machinery/door/airlock/A = D
+					A.locked = 0
+					A.welded = 0
+					A.jammed = 0
+				D.open(1)
+			else
+				if(prob(15))
+					D.visible_message("<span class='warning'>\The [D] creaks open under force, steadily</span>")
+					D.open(1)
+	busy = 0
+	stop_automated_movement = 0
+
+/mob/living/simple_animal/hostile/necro/zombie/proc/check_edibility(var/mob/living/carbon/human/target)
+	if(isjusthuman(target)) //Humans are always edible
+		return 1
+	if(target.health > -400) //So they're not caught eating the same dumb bird all day
+		return 1
+
+	return 0
+
+/mob/living/simple_animal/hostile/necro/zombie/proc/eat(var/mob/living/carbon/human/target)
+	//Deal a random amount of brute damage to the corpse in question, heal the zombie by the damage dealt halved
+	visible_message("<span class='notice'>\The [src] starts to take a bite out of \the [target].</span>")
+	stop_automated_movement = 1
+	if(do_after(src, target, 50, needhand = FALSE))
+		playsound(get_turf(src), 'sound/weapons/bite.ogg', 50, 1)
+		var/damage = rand(melee_damage_lower, melee_damage_upper)
+		target.adjustBruteLoss(damage)
+		health += (damage/2)
+		if(maxHealth < health_cap)
+			maxHealth += 5 //A well fed zombie is a scary zombie
+		times_eaten += 1
+	stop_automated_movement = 0
+	busy = 0
+
+/mob/living/simple_animal/hostile/necro/zombie/proc/check_evolve()
```

Comment:
Because it was a lot longer before comic found out, then I had to split it up to each mob that was evolving.

>ComicIronic said: But it's still breaching OO principles. If you are doing different things based on the type of src, you should overwriting the inherited proc.